testlib/utils.py

```diff
@@ -76,6 +76,29 @@ def process_exists(name):
     return None
 
 
+def terminate_traces(name):
+    """
+    Terminate trace processes with the given filename.  This is
+    intended for Python scripts whose name will be in the cmdline, but
+    not in the process name.
+    :param name: name of script to clean up
+    :type name: str
+    return: None
+    """
+    procs = []
+    for proc in psutil.process_iter(["cmdline"]):
+        try:
+            for param in proc.info["cmdline"]:
```

Comment:
Here's a place where you could get fancy with list comprehension and this time probably increase comprehensibility:

```
if any(name in param for param in proc.info["cmdline"]):
   procs.append(proc)
```

Of course, as we discussed, it probably makes sense just to kill the process as soon as you find it...

Reply:
Also, I think I'ld prefer ``` name == param ``` to ```name in param```. If the name argument is "a", this function could end up killing a lot of processes.

Reply:
Resolved.